scheduler/src/cook/mesos/task.clj

```diff
@@ -111,14 +112,14 @@
 (defn take-all-scalar-resources-for-task
   [resources task]
   (reduce
-   (fn [{:keys [remaining-resources mesos-messages]} [resource-keyword amount]]
-     (let [resource-name (name resource-keyword)
-           adjustment (take-resources remaining-resources resource-name amount)]
-       {:remaining-resources (assoc remaining-resources resource-name (:remaining-resources adjustment))
-        :mesos-messages (into mesos-messages (:mesos-messages adjustment))}))
-   {:remaining-resources resources
-    :mesos-messages []}
-   (-> task :resources vec)))
+    (fn [{:keys [remaining-resources mesos-messages]} [resource-keyword amount]]
+      (let [resource-name (name resource-keyword)
+            adjustment (take-resources remaining-resources resource-name amount)]
+        {:remaining-resources (assoc remaining-resources resource-name (:remaining-resources adjustment))
+         :mesos-messages (into mesos-messages (:mesos-messages adjustment))}))
+    {:remaining-resources resources
+     :mesos-messages []}
+    (.getScalarRequests ^com.netflix.fenzo.TaskRequest (:task-request task))))
```

Comment:
The `:task-request` is constructed differently, and doesn't automatically mirror all the resources that actually need to be debited from the offer. The Fenzo object does all that accounting.